libs/external-router/src/model/type-route-pair.ts

```diff
@@ -0,0 +1,17 @@
+import { RouteWithoutPath } from './route-without-path';
+import { DaffExternalRouterType } from './route-type';
+
+/**
+ * A model used to describe the relationship between a {@link DaffExternalRouterType}
+ * and its corresponding route.
+ *
+ * The TypeRoutePair is the pre-provided "glue" that a developer will provide
+ * ahead-of-time so that that the External Router knows what the list of available
+ * "types" are (along with that type's corresponding route).
+ *
+ * @docs-private
+ */
+export interface TypeRoutePair {
+	type: DaffExternalRouterType;
+	route: RouteWithoutPath;
```

Comment:
Something I'm not quite understanding is why this is typed as `RouteWithoutPath`. I would've thought it was just some string that the platform needs to determine if you're requesting a product or a category or whatever in case there are url collisions between those feature areas. But that must not be how it works. What will this "route" be in this object and when will it be provided?

Reply:
I renamed the `DaffExternalRouterType` to the `DaffExternalRouteType` for clarity here. I think this will also become more clear once published to npm so that I can make a stackblitz.